autorest/models/name_converter.py

```diff
@@ -0,0 +1,186 @@
+import re
+from ..common.python_mappings import basic_latin_chars, reserved_words
+
+class NameConverter:
+
+    def __init__(self):
+        self.seen_yaml_ids = []
```

Comment:
why?

Reply:
Looking at the code, it seems you're trying to solve the recursive reference, but as we discussed it's not the best way. Create the "python" dict, and fill it afterwards if necessary.

Reply:
this was so I wouldn't convert the same chunks multiple times, not so much to avoid recursion. I've changed it now so I check to see if the yaml_data at ['language'] already has 'python'. If so, it's already been converted so I dont' need to do anything